assets/js/base/components/cart-checkout/shipping-rates-control-package/index.tsx

```diff
@@ -66,13 +70,25 @@ export const ShippingRatesControlPackage = ( {
 	showItems = false,
 }: PackageProps ): ReactElement => {
 	const { selectShippingRate } = useSelectShippingRate();
+	const data = __experimentalApplyCheckoutFilter( {
+		filterName: 'itemName',
+		defaultValue: packageData.name,
+	} );
+	const ALLOWED_TAGS = [ 'a', 'b', 'em', 'i', 'strong', 'p', 'br' ];
+	const ALLOWED_ATTR = [ 'target', 'href', 'rel', 'name', 'download' ];
+	const sanitizeHTML = ( html: string ) => {
+		return {
+			__html: sanitize( html, { ALLOWED_TAGS, ALLOWED_ATTR } ),
+		};
+	};
```

Comment:
This piece of code will be instrumental! I think we will be using it every time we need to output safe HTML. Could we make it reusable? Maybe it can live in `assets/js/utils` (`@woocommerce/utils`) or `assets/js/base/utils` (`@woocommerce/base-utils`). We can name this file `sanitize-html.ts`, for example! 🙌 

```tsx
const ALLOWED_TAGS = [ 'a', 'b', 'em', 'i', 'strong', 'p', 'br' ];
	const ALLOWED_ATTR = [ 'target', 'href', 'rel', 'name', 'download' ];
	const sanitizeHTML = ( html: string ) => {
		return {
			__html: sanitize( html, { ALLOWED_TAGS, ALLOWED_ATTR } ),
		};
	};
```

Reply:
@nielslange I think it's [the same code as mine](https://github.com/woocommerce/woocommerce-blocks/pull/7145),  It seems like it's working in this PR, which part do you think is not working?

Reply:
> I think we will be using it every time we need to output safe HTML. Could we make it reusable? Maybe it can live in assets/js/utils (@woocommerce/utils) or assets/js/base/utils (@woocommerce/base-utils). We can name this file sanitize-html.ts

Sounds good!

@nielslange Since it's working fine, I've merged my PR.  Would you like to move this to the utils file?

Reply:
> @nielslange I think it's [the same code as mine](https://github.com/woocommerce/woocommerce-blocks/pull/7145), It seems like it's working in this PR, which part do you think is not working?

@hsingyuc The default sanitisation is working fine. But looking at the example on https://github.com/cure53/DOMPurify#can-i-configure-dompurify, I see the following example:

```js
// allow only <b> and <q> with style attributes
var clean = DOMPurify.sanitize(dirty, {ALLOWED_TAGS: ['b', 'q'], ALLOWED_ATTR: ['style']});
```

Looking at your implementation, I'd say the code is executed as:

```js
const sanitizeHTML = ( html: string ) => {
  return {
    __html: sanitize( html, { [ 'a', 'b', 'em', 'i', 'strong', 'p', 'br' ],  [ 'target', 'href', 'rel', 'name', 'download' ] } ),
  };
```

As both `[ 'a', 'b', 'em', 'i', 'strong', 'p', 'br' ]` and `[ 'target', 'href', 'rel', 'name', 'download' ]` are unnamed objects, I believe that `DOMPurify.sanitize()` will ignore these values and will execute the code as:

```js
const sanitizeHTML = ( html: string ) => {
  return {
    __html: sanitize( html ),
  };
```

When working on this PR, I changed 
```js
const ALLOWED_TAGS = [ 'a', 'b', 'em', 'i', 'strong', 'p', 'br' ];
```
to
```js
const ALLOWED_TAGS = [ 'b', 'em', 'i', 'strong', 'p', 'br' ];
```
meaning `a` is no longer part of the `ALLOWED_TAGS`. Yet, the anchor element was still visible and did not get removed.

> > I think we will be using it every time we need to output safe HTML. Could we make it reusable? Maybe it can live in assets/js/utils (@woocommerce/utils) or assets/js/base/utils (@woocommerce/base-utils). We can name this file sanitize-html.ts
> 
> Sounds good!
> 
> @nielslange Since it's working fine, I've merged my PR. Would you like to move this to the utils file?

I've created the utils function last Friday, but got sidetracked due to p1663920815156619-slack-C8X6Q7XQU. I'll finalise this PR today, which also moves the sanitisation into a reusable component. I'll let you know once I'm done with that, so that you can use the reusable component as well.

Reply:
> This piece of code will be instrumental! I think we will be using it every time we need to output safe HTML. Could we make it reusable? Maybe it can live in `assets/js/utils` (`@woocommerce/utils`) or `assets/js/base/utils` (`@woocommerce/base-utils`). We can name this file `sanitize-html.ts`, for example! 🙌

Thanks for suggesting this, @tarhi-saad. I just created `sanitize` as a reusable util and made it available via `@woocommerce/utils`.

Reply:
```
const sanitizeHTML = ( html: string ) => {
  return {
    __html: sanitize( html, { [ 'a', 'b', 'em', 'i', 'strong', 'p', 'br' ],  [ 'target', 'href', 'rel', 'name', 'download' ] } ),
  };
```

@nielslange I don't know if you saw my [previous comment](https://github.com/woocommerce/woocommerce-blocks/pull/7145#discussion_r978890021) but this isn't what happens.  This would throw an error because an object can't be created without keys.  For example, try this in your browser:

```
const myProperty = ['a', 'b', 'c' ];
const myObj = { myProperty };
console.log( myObj );
```

This will output:

```
{
    "myProperty": [ "a", "b", "c" ]
}
```

So I suspect something else was happening in your testing.  Maybe the files weren't rebuilt after you removed the anchor tag.

>Would you like to move this to the utils file?

It looks like you've merged this code.  Is there any reason you did not move my method to the utils file?  Or should I open an issue to clean this up so we only have one sanitize method?

Reply:
> @nielslange I don't know if you saw my [previous comment](https://github.com/woocommerce/woocommerce-blocks/pull/7145#discussion_r978890021) but this isn't what happens. This would throw an error because an object can't be created without keys. For example, try this in your browser:
> 
> ```
> const myProperty = ['a', 'b', 'c' ];
> const myObj = { myProperty };
> console.log( myObj );
> ```
> 
> This will output:
> 
> ```
> {
>     "myProperty": [ "a", "b", "c" ]
> }
> ```
> 
> So I suspect something else was happening in your testing. Maybe the files weren't rebuilt after you removed the anchor tag.
> 
> >

I've seen your comment, but replied here to avoid that we're discussing the same topic on two different places. Thanks for adding more context and showing the example. Apparently, something went wrong on my end, and I was able to reproduce your result.

> Or should I open an issue to clean this up so we only have one sanitize method?

I see that you already created the PR https://github.com/woocommerce/woocommerce-blocks/pull/7231, so I'll leave this up to you.